app/src/pages/playground/chatTemplateDndSensors.ts

```diff
@@ -0,0 +1,85 @@
+import { Draggable, KeyboardSensor, PointerSensor } from "@dnd-kit/dom";
+
+/**
+ * Copy of @dnd-kit/dom
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L18
+ */
+type KeyCode = KeyboardEvent["code"];
+/**
+ * Copy of @dnd-kit/dom KeyboardCodes type.
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L20
+ */
+type KeyboardCodes = {
+  start: KeyCode[];
+  cancel: KeyCode[];
+  end: KeyCode[];
+  up: KeyCode[];
+  down: KeyCode[];
+  left: KeyCode[];
+  right: KeyCode[];
+};
+
+/**
+ * Copy of @dnd-kit/dom KeyboardSensorOptions type.
+ * @see https://github.com/clauderic/dnd-kit/blob/experimental/packages/dom/src/core/sensors/keyboard/KeyboardSensor.ts#L30
+ */
+interface KeyboardSensorOptions {
+  keyboardCodes?: KeyboardCodes;
+}
+
+/**
+ * A dnd-kit {@link KeyboardSensor} that ignores keyboard events when the user is typing in a text input.
+ */
+export class ChatTemplateKeyboardSensor extends KeyboardSensor {
+  protected handleStart(
+    event: KeyboardEvent,
+    source: Draggable,
+    options: KeyboardSensorOptions | undefined
+  ): void {
+    /**
+     * In some instances a text input may be nested within a draggable element.
+     * In that case we want to ignore the keyboard event to allow the user to type.
+     * In all other cases, we want to allow the user to move the draggable element.
+     */
+    if (
```

Comment:
This is still preventing us from opening the role picker. Testing more and more targets feels brittle or easy to forget tbh but I'm not sure what else we try

Reply:
oh shoot i didn't test that, thanks for pulling down, let me see what i can do